client/src/components/ImgBox.jsx

```diff
@@ -1,53 +1,56 @@
-import React, { useState } from "react";
+import React, { useState, useCallback } from "react";
 import "./ImgBox.css";
 
-const ImgBox = ({ onChange, id, color, opacity }) => { 
+const ImgBox = ({ onChange, id, color, opacity }) => {
   const [image, setImage] = useState(null);
 
-  const handleImageUpload = (e) => {
+  const handleImageUpload = useCallback((e) => {
     const file = e.target.files[0];
     if (file) {
       const reader = new FileReader();
       reader.onload = (e) => {
-        setImage(e.target.result);
+        const newImage = e.target.result;
+        setImage(newImage);
+        onChange(id, newImage);
       };
       reader.readAsDataURL(file);
     }
-  };
-
+  }, [onChange, id]);
 
   return (
-    <div className="imgbox-wrapper" style={{ backgroundColor: color, opacity: opacity / 100 }}>
-      <div className="imgbox-content">
+    <div
+      className="imgbox-wrapper"
+      style={{ backgroundColor: color, opacity: opacity / 100 }}
+    >
+      <input
+        id={`file-input-${id}`}
+        type="file"
+        accept="image/*"
```

Comment:
this would mean we can accept any image type maybe we want to be a bit more restrictive here?